archunit/src/main/java/com/tngtech/archunit/library/modules/ArchModule.java

```diff
@@ -0,0 +1,305 @@
+    public String getName() {
+        return descriptor.getName();
+    }
+
+    /**
+     * @return The {@link ArchModule.Descriptor} of this {@link ArchModule}
+     */
+    @PublicAPI(usage = ACCESS)
+    public DESCRIPTOR getDescriptor() {
+        return descriptor;
+    }
+
+    /**
+     * @return All {@link Dependency dependencies} where the {@link Dependency#getOriginClass() origin class}
+     *         is contained within this {@link ArchModule}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public Set<Dependency> getClassDependenciesFromSelf() {
+        return classDependenciesFromSelf;
+    }
+
+    /**
+     * @return All {@link Dependency dependencies} where the {@link Dependency#getTargetClass() target class}
+     *         is contained within this {@link ArchModule}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public Set<Dependency> getClassDependenciesToSelf() {
+        return classDependenciesToSelf.get();
+    }
+
+    /**
+     * @return All {@link ModuleDependency module dependencies} where the {@link ModuleDependency#getOrigin() origin}
+     *         equals this {@link ArchModule}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public Set<ModuleDependency<DESCRIPTOR>> getModuleDependenciesFromSelf() {
+        return moduleDependenciesFromSelf;
+    }
+
+    /**
+     * @return All {@link ModuleDependency module dependencies} where the {@link ModuleDependency#getTarget() target}
+     *         equals this {@link ArchModule}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public Set<ModuleDependency<DESCRIPTOR>> getModuleDependenciesToSelf() {
+        return moduleDependenciesToSelf;
+    }
+
+    @PublicAPI(usage = ACCESS)
+    public Set<Dependency> getUndefinedDependencies() {
+        return undefinedDependencies;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(identifier);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        if (!super.equals(obj)) {
+            return false;
+        }
+        final ArchModule<?> other = (ArchModule<?>) obj;
+        return Objects.equals(this.identifier, other.identifier);
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "{identifier=" + getIdentifier() + ", name=" + getName() + '}';
+    }
+
+    /**
+     * An {@link Identifier} of an {@link ArchModule}. An {@link Identifier} is basically an ordered list of string parts that
+     * uniquely identifies an {@link ArchModule}, i.e. two {@link ArchModule modules} are equal, if and only if
+     * their identifier is equal (i.e. all the textual parts of the identifier match in order).
+     */
+    @PublicAPI(usage = ACCESS)
+    public static final class Identifier implements Iterable<String> {
+        private final List<String> parts;
```

Comment:
I've probably missed something, but my impression was that the Identifier API with multiple parts might be more powerful than actually needed. Wouldn't a `String` usually be good enough?

Reply:
This comes a little from the package parsing. Suppose I'm parsing hierarchical packages like
```
my.app.order.wares
my.app.order.billing
my.app.customer.management
my.app.customer.contact
```
with `my.app.(*).(*)..`. Then the structured identifier would make it quite easy to derive a name like `Module Order - Submodule Wares` from this. It would of course also be possible to parse a single string identifier again, but this API seemed more natural from this viewpoint :man_shrugging: 

Does this make sense to you or do you think I've just given too much emphasis to a corner case, complicating the typical use case? :thinking:

Reply:
The only _"complications"_ are that users have to choose from a richer API, and that maintainers have to, well, maintain it. 😜

Reply:
I hereby accept the risk :wink: (you can hold me accountable in a year or so when we're bitten by this :stuck_out_tongue_winking_eye:)